aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -460,13 +599,91 @@ Tensor max_unpooling2d_backward_cpu(
     const Tensor& grad_output,
     const Tensor& self,
     const Tensor& indices,
-    IntArrayRef output_size) {
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
   auto grad_input = at::empty_like(self, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   max_unpooling2d_backward_out_cpu(
-      grad_input, grad_output, self, indices, output_size);
+      grad_input, grad_output, self, indices, output_size, stride, padding);
+  return grad_input;
+}
+
+Tensor& max_unpooling1d_backward_out_cpu(
+    Tensor& grad_input,
+    const Tensor& grad_output_,
+    const Tensor& self,
+    const Tensor& indices_,
+    IntArrayRef output_size,
+    IntArrayRef stride,
+    IntArrayRef padding) {
+  TORCH_CHECK(grad_input.is_contiguous(), "grad_input must be contiguous");
+  int64_t oheight = output_size[0];
+  int64_t owidth = 1;
+  int dimw = 2;
+  int dimh = 1;
+  int nbatch = 1;
+  int nslices;
+  int iheight;
+  int iwidth;
+
+  max_unpooling1d_shape_check(
+      self, grad_output_, indices_, output_size, stride, padding);
+  /* get contiguous gradOutput and indices and extend on dimension */
+  auto grad_output = grad_output_.contiguous().unsqueeze(3);
+
+  /* extend one dimension for tensor so that it can use max_unpooling2d_backward_out_cpu_frame method. */
+  auto indices = indices_.contiguous().unsqueeze(3);
+  auto self_ = self.contiguous().unsqueeze(3);
+
+  /* resize */
+  grad_input.resize_as_(self_);
+  grad_input.zero_();
+
+  if (self_.ndimension() == 4) {
```

Comment:
Same issue with these additions as above. use an if/else for the number of dimensions to set these properly.